haystack/nodes/file_converter/pdf.py

```diff
@@ -15,9 +26,10 @@ def __init__(
         remove_numeric_tables: bool = False,
         valid_languages: Optional[List[str]] = None,
         id_hash_keys: Optional[List[str]] = None,
-        encoding: Optional[str] = "UTF-8",
-        keep_physical_layout: bool = False,
-    ):
+        encoding: Optional[str] = None,
+        keep_physical_layout: Optional[bool] = None,
```

Comment:
Let's deprecate this parameter as well and instead have a parameter `sort_by_position` that would be passed on to PyMuPDF's `sort` parameter. A doc string for `sort_by_position` could be the following:
```
Specifies whether to sort the extracted text by positional coordinates or logical reading order.
    If set to True, the text is sorted first by vertical position, and then by horizontal position.
    If set to False (default), the logical reading order in the PDF is used.
```